usb/esp_tinyusb/include/tusb_config.h

```diff
@@ -82,7 +82,14 @@ extern "C" {
 #   define CONFIG_TINYUSB_DEBUG_LEVEL 0
 #endif
 
-#define CFG_TUSB_RHPORT0_MODE       OPT_MODE_DEVICE | OPT_MODE_FULL_SPEED
+#ifdef CONFIG_TINYUSB_RHPORT_HS
+#   define CFG_TUSB_RHPORT1_MODE    OPT_MODE_DEVICE | OPT_MODE_HIGH_SPEED
+#   define CFG_TUSB_EP_SIZE         512
```

Comment:
When building an HS device (with HS MPS), do you know if the EP size gets set back to 64 if this HS device is plugged into a FS Host?

IMO it's not super critical since its unlikely our initial demos will be plugged into an FS host, so we could always leave this as a follow up.

Reply:
That is exactly the question was mentioned a couple of minutes before.
I am going to check it.

Reply:
@tore-espressif @Dazza0 Yep, to be able to work on both hosts (FS and HS) we should be able to provide a correct configuration descriptor with the EP size, which should be 64 or 512.
Right now, we don't have an option to register two configuration descriptor, only one of them via esp_tinyusb: 
`const uint8_t *configuration_descriptor;   /*!< Pointer to a configuration descriptor. If set to NULL, TinyUSB device will use a default configuration descriptor whose values are set in Kconfig */`

But in P4 we need to register both of them and change the EP size based on the speed.

```
// Invoked when received GET CONFIGURATION DESCRIPTOR
// Application return pointer to descriptor
// Descriptor contents must exist long enough for transfer to complete
uint8_t const * tud_descriptor_configuration_cb(uint8_t index)
{
  (void) index; // for multiple configurations

#if TUD_OPT_HIGH_SPEED
  // Although we are highspeed, host may be fullspeed.
  return (tud_speed_get() == TUSB_SPEED_HIGH) ?  desc_hs_configuration : desc_fs_configuration;
#else
  return desc_fs_configuration;
#endif
}
```
I guess right now I can revert the changes to configure EP size based on speed and leave only the option to set the ep size in examples code. 
After that, we could think/discuss how we want to achieve the goal of config decs for FS and HS: register two const descriptors or just implement a callback.

Reply:
Yep, otherwise, right now (if always return FS config descriptor with EP size 64) then S3 MSC Host + P4 MSC Device are working in pair.

Reply:
@tore-espressif @Dazza0 I have removed the HS configuration descriptor from current PR. 

Right now the HS P4 Device won't work in any FS Host port. Compatibility only with HS Hosts.